tests/gordo_components/builder/test_builder.py

```diff
@@ -262,6 +294,174 @@ def test_get_metadata_helper(model: BaseEstimator, expect_empty_dict: bool):
+        gordo_components.model.anomaly.diff.DiffBasedAnomalyDetector:
+            scaler: sklearn.preprocessing.data.MinMaxScaler
+            base_estimator:
+                sklearn.compose.TransformedTargetRegressor:
+                    transformer: sklearn.preprocessing.data.MinMaxScaler
+                    regressor:
+                        sklearn.pipeline.Pipeline:
+                            steps:
+                            - sklearn.preprocessing.data.MinMaxScaler
+                            - gordo_components.model.models.KerasAutoEncoder:
+                                kind: feedforward_hourglass
+                                batch_size: 3
+                                compression_factor: 0.5
+                                encoding_layers: 1
+                                func: tanh
+                                out_func: linear
+                                epochs: 1
+        """
+    raw_model_config2 = f"""
+       sklearn.compose.TransformedTargetRegressor:
+           transformer: sklearn.preprocessing.data.MinMaxScaler
+           regressor:
+              sklearn.pipeline.Pipeline:
+                  steps:
+                  - sklearn.preprocessing.data.MinMaxScaler
+                  - gordo_components.model.models.KerasAutoEncoder:
+                      kind: feedforward_hourglass
+                      batch_size: 2
+                      compression_factor: 0.5
+                      encoding_layers: 1
+                      func: tanh
+                      out_func: linear
+                      epochs: 1
+                     """
+    raw_model_config3 = f""" 
+              sklearn.pipeline.Pipeline:
+                  steps:
+                  - sklearn.preprocessing.data.MinMaxScaler
+                  - gordo_components.model.models.KerasAutoEncoder:
+                      kind: feedforward_hourglass
+                      batch_size: 2
+                      compression_factor: 0.5
+                      encoding_layers: 1
+                      func: tanh
+                      out_func: linear
+                      epochs: 1
+                     """
+    model_config1 = yaml.load(raw_model_config1, Loader=yaml.FullLoader)
+    model_config2 = yaml.load(raw_model_config2, Loader=yaml.FullLoader)
+    model_config3 = yaml.load(raw_model_config3, Loader=yaml.FullLoader)
+
+    model1, metadata1 = build_model(
+        name="model-name",
+        model_config=model_config1,
+        data_config=data_config,
+        metadata={},
+    )
+    model2, metadata2 = build_model(
+        name="model-name",
+        model_config=model_config2,
+        data_config=data_config,
+        metadata={},
+    )
+    model3, metadata3 = build_model(
+        name="model-name",
+        model_config=model_config3,
+        data_config=data_config,
+        metadata={},
+    )
+
+    metadata_check(metadata1, False)
+    metadata_check(metadata2, False)
+    metadata_check(metadata3, False)
```

Comment:
This test, specifically the `raw_model` configs, should be parameterized `pytest.mark.parameterize`